client/src/App.js

```diff
@@ -3,31 +3,143 @@ import { Switch, Route } from 'react-router-dom';
 import { saveToLocalStorage, loadFromLocalStorage } from './lib/localStorage';
 import styled from 'styled-components/macro';
 import Home from './pages/Home.js';
-import BurgerMenu from './BurgerMenu.js';
+import BurgerMenu from './components/BurgerMenu.js';
 import MarketCard from './pages/MarktCard.js';
 import MarketForm from './pages/MarktForm.js';
 
 function App() {
   const [markets, setMarkets] = useState(loadFromLocalStorage('Markets') ?? []);
+  const [bookmarkedMarkets, setBookmarkedMarkets] = useState(
+    loadFromLocalStorage('bookmarkedMarkets') ?? []
+  );
+  const [filteredMarkets, setFilteredMarkets] = useState([]);
 
   function addComment(comment, marketToUpdate) {
-    const updatedMarkets = markets.map((market) => {
-      // @TODO: Replace .name with ._id later when markets are fetched from API / DB
-      if (market.name === marketToUpdate.name) {
+    const upToDateMarkets = markets.filter(
+      (market) => market._id !== marketToUpdate._id
+    );
+    markets.map((market) => {
+      if (market._id === marketToUpdate._id) {
         market.comments.push(comment);
+        fetch('http://localhost:4000/market/' + marketToUpdate._id, {
+          method: 'PUT',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(marketToUpdate),
+        })
+          .then((result) => result.json())
+          .then((updatedMarket) => {
+            setMarkets([...upToDateMarkets, updatedMarket]);
+          })
+          .catch((error) => console.error(error));
       }
       return market;
     });
+  }
 
-    setMarkets(updatedMarkets);
+  function addRating(rating, marketToUpdate) {
+    const upToDateMarkets = markets.filter(
+      (market) => market._id !== marketToUpdate._id
+    );
+    markets.map((market) => {
+      if (market._id === marketToUpdate._id) {
+        market.rating.push(rating);
+        fetch('http://localhost:4000/market/' + marketToUpdate._id, {
+          method: 'PUT',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(marketToUpdate),
+        })
+          .then((result) => result.json())
+          .then((updatedMarket) => {
+            setMarkets([...upToDateMarkets, updatedMarket]);
+          })
+          .catch((error) => console.error(error));
+      }
+      return market;
+    });
   }
 
+  useEffect(() => {
+    fetch('http://localhost:4000/market')
+      .then((result) => result.json())
+      .then((marketFromApi) => setMarkets(marketFromApi))
+      .catch((error) => console.error(error));
+  }, []);
+
+  useEffect(() => {
+    setFilteredMarkets(markets);
+  }, [markets]);
+
   useEffect(() => {
     saveToLocalStorage('Markets', markets);
   }, [markets]);
 
+  useEffect(() => {
```

Comment:
Your structure is a bit of a mixture of useEffects and functions. 
I would recommend that you put all hooks on top of your component (`useState` and `useEffect`) .
Afterwards the functions can follow. 
But also consider extracting the functions from the file and importing them :)